### DISCUSSION TITLE
I am new to or-tools

### DISCUSSION THREAD

**Discussion by mtoczycki** [google/or-tools#3053] [Routing (and legacy CP) questions]
I am trying to find out the best way to figure out how to make sure that an origin does not stay on a vehicle for too long. For example, one origin has a time limit of 60 minutes time limit on the vehicle while the other origin has 30 minutes time limit on the vehicle.  I am thinking maybe time windows, but I think dimensions would be best, I need to do this for each origin.  I have been following the examples so far on or-tools with java and have the link here: https://developers.google.com/optimization/routing/penalties#java_3 the only thing that I have really changed that is important is the print and that the depot row is all 0 and this distance matrix is in terms of seconds of time.  I am using the java version, any suggestions?

The question I have is if I have a bus, how can I limit that the person is not on the bus for more than an x amount of time per index or origin.  I am thinking that maybe I have to have as many dimensions as there are indices, but I am having a hard time figuring out how to code this.

**Comment by lperron** [comment]
Instead of apologizing in the title, you should have an informative title. Then include the code of a complete mini-example that just misses the relevant constraint.

Furthermore, if you model this as PDP with a time dimension, it is easy to constrain the cumul vars of the pickup and delivery for the 'person' to be close enough.
